app/views/govuk_publishing_components/components/_search.html.erb

```diff
@@ -8,16 +8,21 @@
   value ||= ""
   id ||= "search-main-" + SecureRandom.hex(4)
   label_text ||= "Search GOV.UK"
+  show_search_button = true if local_assigns.exclude?(:show_search_button)
```

Comment:
I'd be tempted to rewrite this as

```
hide_search_button ||= false
...

<% unless hide_search_button %>
...
```

Maybe makes it a bit cleaner and more readable?

Reply:
Yeah, I wasn't entirely happy with my approach because I tried `show_search_button ||= true` and this failed when I tried to set the variable to false.

I prefer your example.